public/src/overmind/actions.tsx

```diff
@@ -190,84 +215,86 @@ export const updateEnrollment = async ({ state, effects }: Context, { enrollment
-            // If successful, update enrollment in state with new status
-            if (status === Enrollment_UserStatus.NONE) {
-                // If the enrollment is rejected, remove it from state
-                enrollments.splice(found, 1)
-            } else {
-                // If the enrollment is accepted, update the enrollment in state
-                enrollments[found].status = status
-            }
+    // Send updated enrollment to server
+    const response = await effects.api.client.updateEnrollments({ enrollments: [temp] })
+    if (success(response)) {
+        // If successful, update enrollment in state with new status
+        if (status === Enrollment_UserStatus.NONE) {
+            // If the enrollment is rejected, remove it from state
+            enrollments.splice(found, 1)
+        } else {
+            // If the enrollment is accepted, update the enrollment in state
+            enrollments[found].status = status
         }
     }
 }
 
 /** approvePendingEnrollments approves all pending enrollments for the current course */
 export const approvePendingEnrollments = async ({ state, actions, effects }: Context): Promise<void> => {
-    if (confirm("Please confirm that you want to approve all students")) {
-        // Clone and set status to student for all pending enrollments.
-        // We need to clone the enrollments to avoid modifying the state directly.
-        // We do not want to update set the enrollment status before the update is successful.
-        const enrollments = state.pendingEnrollments.map(e => {
-            const temp = e.clone()
-            temp.status = Enrollment_UserStatus.STUDENT
-            return temp
-        })
+    if (!confirm("Please confirm that you want to approve all students")) {
+        return
+    }
 
-        // Send updated enrollments to server
-        const response = await effects.grpcMan.updateEnrollments(enrollments)
-        if (success(response)) {
-            for (const enrollment of state.pendingEnrollments) {
-                enrollment.status = Enrollment_UserStatus.STUDENT
-            }
-        } else {
-            // Fetch enrollments again if update failed in case the user was able to approve some enrollments
-            await actions.getEnrollmentsByCourse({ courseID: state.activeCourse, statuses: [Enrollment_UserStatus.PENDING] })
-            actions.alertHandler(response)
+    // Clone and set status to student for all pending enrollments.
+    // We need to clone the enrollments to avoid modifying the state directly.
+    // We do not want to update set the enrollment status before the update is successful.
+    const enrollments = state.pendingEnrollments.map(e => {
+        const temp = e.clone()
+        temp.status = Enrollment_UserStatus.STUDENT
+        return temp
+    })
+
+    // Send updated enrollments to server
+    const response = await effects.api.client.updateEnrollments({ enrollments })
+    if (success(response)) {
```

Comment:
Suggest to flip the if, and use the common pattern `if (response.error)` and keep the happy path non-indented after the `await ... ` and a return.